csrc/scheduler/matmul.cpp

```diff
@@ -628,7 +630,19 @@ void scheduleFusionInputsForEpilogue(const mma_utils::RolesMap& roles_map) {
 
     scheduler_utils::BoundedDirectionalTransformPropagator::backward(
         output_d, -1, c_tvs);
-    scheduler_utils::parallelizeAllLike(output_d, -1, cached_tvs);
+
+    std::unordered_set<ParallelType> parallel_types = {};
+    if (with_smem_epilogue) {
```

Comment:
Instead of guarding with `with_smem_epilogue` and turning off vectorization for all tensors, I think the standard way for us to determine which input/output is vectorizable is in:

https://github.com/NVIDIA/Fuser/blob/280c3186112bf027f12189d9a7404501609965f7/csrc/scheduler/pointwise.cpp#L164-L171

This logic is used by all schedulers.

Reply:
You are right, in the end matmul scheduler must use vectorization analysis.

The changes I've prepared are meant to resolve immediate issue where matmul scheduler allows generating kernels that have issues that are not only limited to Ada architecture. I mentioned this in the comment in the code branch, the follow up action should be adding a support for vectorization analysis.

I will look into adding the vectorization, along with heuristics cache through `HeuristicsSummaryEntry`, but this means that the issue will be present in the system for longer time.

Reply:
I think it is totally up to you to decide how you would like to proceed. I am OK with having this hack now, and resolve it in the future. Before delivering matmul to any user, it is totally up to us to decide how we approach all the required features.

On the other hand, bringing up vectorization analysis does not have to be an all-or-nothing task either. One incremental task we can do is: We can just copy-paste the above code, and the beta tensor will be in `vectorizable_inputs_outputs_entry`, but the bias will not. Then with this information, we can first do a `parallelizeAllLike(allParallelTypesExcept({ParallelType::Vectorize}))` then followed with a `parallelizeAllLike({ParallelType::Vectorize})` on only vectorizable outputs and input caches except the input caches for A, B and the non-vectorizable inputs. We can still just naively vectorize 8 like what we are doing today, and only use `vectorizable_inputs_outputs_entry` to separate bias with other vectorizable tensor. This will resolve the concern @liqiangxl have in https://github.com/NVIDIA/Fuser/pull/707#discussion_r1290271990

Reply:
Thank you for sharing details how to approach the problem!

The plan now looks like this:
- the fix for smem & epilogue like bias will be promoted, as a result reading of the additional epilogue inputs won't be vectorized
- I will prepare a PR with a follow up feature, strided batch gemm
- when bgemm PR is promoted to `main` I will get back to vectorization issue for which this PR introduces a WAR

@liqiangxl for visibility